app/assets/javascripts/course/assessment/submission/answer/programming_submit.js

```diff
@@ -27,13 +27,15 @@
     var action = $form.attr('action');
     var method = $form.attr('method');
 
-    var data = $form.serializeArray();
-    data.push({name: 'attempting_answer_id', value: answerId});
+    var data = new FormData($form[0]);
+    data.append('attempting_answer_id', answerId);
     $.ajax({
       url: action,
       method: method,
-      data: $.param(data),
-      dataType: 'json',
+      data: data,
+      enctype: 'multipart/form-data',
+      processData: false,
```

Comment:
would be great if we only enable `multipart` when there are files in the params, so that there's less overhead for all other types of answers.